### DISCUSSION TITLE
Feature Requests

### DISCUSSION THREAD

**Discussion by bennyguo** [threestudio-project/threestudio#46] [Announcements]
**Hello, 3D enthusiasts!**

I know that many of you are expecting features/models from threestudio, but since we're currently a very small team, it could be hard to meet everyone's needs. So I set up this discussion to find the most-wanted features/models.

- **Feel free to reply with your requests, but remember to check if there're already similar requests to avoid duplications.**
- **Hit the upvote button for the features/models you want. Every once in a while we'll select some of the most upvoted requests and see if we can get them done.**

I'll give an example to start this discussion:

**Comment by bennyguo** [comment]
Implementation of [DreamAvatar](https://yukangcao.github.io/DreamAvatar) for high-quality 3D avatar generation from text prompts.

![image](https://github.com/threestudio-project/threestudio/assets/19284678/978d5aed-6191-4f42-95b8-bf8ce843ba29)

**Reply by andorxornot** [reply]
it is interesting that they use a parameterized human model.

**Reply by yuedajiong** [reply]
The biggest one is:  3+ image(s) guided game avatar generator.    interactive ui/drag,etc or text guide edit should be better.
Any plan to implements?   such as sparsefusion.

**Comment by qtabdullah** [comment]
Image to 3D

**Comment by andorxornot** [comment]
Implementation of [Perp-Neg-stablediffusion](https://github.com/Perp-Neg/Perp-Neg-stablediffusion) ,  this seems to solve the Janus problem

**Reply by thuliu-yt16** [reply]
Already in our plan :)

**Reply by bennyguo** [reply]
Supported! Please follow the instructions [here](https://github.com/threestudio-project/threestudio#tips-on-improving-quality) to try it out.

**Reply by IkuStudies** [reply]
Nice work!

**Comment by XinYu-Andy** [comment]
Implementation of [TextMesh](https://arxiv.org/abs/2304.12439) , which utilizes pseudo image supervision.

**Reply by bennyguo** [reply]
Implemented! Please follow the instructions [here](https://github.com/threestudio-project/threestudio#textmesh-) to give it a try.

**Comment by qtabdullah** [comment]
A galley

**Comment by LiamSmyth** [comment]
Just an idea, but it would be cool to have a way to preview training live like the instantNERF viewer or NERFStudio?

Longer term, assuming this stuff gets good enough start spending time art directing, it would be cool to be able to watch it update, and if it starts "growing" in a direction you don't like, you can reset it to the last saved checkpoint and tune the training parameters or seed, or even spawn a bunch of evolutionary branches!

**Comment by alievrusik** [comment]
Can you please add this paper method ([TextMesh](https://fabi92.github.io/textmesh/)). Results look really nice (though could be cherypicked) and authors claim that their model beats DreamFusion.

**Reply by bennyguo** [reply]
Implemented! Please follow the instructions [here](https://github.com/threestudio-project/threestudio#textmesh-) to give it a try.

**Comment by claforte** [comment]
Reproduce https://github.com/ashawkey/stable-dreamfusion/issues/263

(My team at Stability AI might help with that. ;-) )

**Comment by bluestyle97** [comment]
Implementation of [ProlificDreamer](https://ml.cs.tsinghua.edu.cn/prolificdreamer/)

**Reply by bennyguo** [reply]
Done. Following [the instruction](https://github.com/threestudio-project/threestudio#prolificdreamer-) to have a try!

**Reply by bluestyle97** [reply]
> Done. Following [the instruction](https://github.com/threestudio-project/threestudio#prolificdreamer-) to have a try!

Wow, so efficient!

**Comment by huthut001** [comment]
Can you improve your model support building 3d model shape or 3d part assembly?

**Comment by PongpreechaSuea** [comment]
Implementation of Neuralangelo: High-Fidelity Neural Surface Reconstruction (https://research.nvidia.com/labs/dir/neuralangelo/) 🥺🥺

**Reply by bennyguo** [reply]
Hi! If you simply want to use tricks from Neuralangelo to train from multi-view images, you can try my [instant-nsr-pl codebase](https://github.com/bennyguo/instant-nsr-pl). I'm also planning to try these tricks in threestudio and see if we can get any benefit from it:)

**Reply by PongpreechaSuea** [reply]
Wow , I've tried your repository and I feel it's very good 
Thanks Implementation of Neuralangelo

**Reply by bennyguo** [reply]
We used techniques from Neuralangelo in our implementation of TextMesh. Welcome to give it a try following [the instructions](https://github.com/threestudio-project/threestudio#textmesh-)!

**Comment by ericwudocomoi** [comment]
[HiFA: High-fidelity Text-to-3D with Advanced Diffusion Guidance](https://hifa-team.github.io/HiFA-site/)

**Comment by qtabdullah** [comment]
Implementation of [ATT3D: Amortized Text-To-3D Object Synthesis](https://research.nvidia.com/labs/toronto-ai/ATT3D/)

**Reply by bennyguo** [reply]
On it :)

**Reply by beibeirory** [reply]
I'm waiting for the implementation of ATT3D sincerely. @bennyguo

**Reply by qtabdullah** [reply]
me too

**Reply by qtabdullah** [reply]
are u done yet?

**Comment by IkuStudies** [comment]
Guys u r crushing it.  I would love to request a feature into threestudio- and that is to have the ability to start the training from a pre existing obj file to guide the shape a bit.

Like a sphere obj file could begin the training and become an apple

While a cylinder might start and form a banana

Or if you had a type of couch as an obj file and wanted to start from that shape to make it spaceaged or something...

Or to start with a sculpture obj file thats frowning and caricaturize it and make the 3d model wildly smiling with a prompt

Maybe this can be done but i would find it incredibly useful to tweak existing 3d models by prompt.  

Btw great work guys been following dreamfusion since its release and amazed by the progress- about to write an article just about the progress

**Reply by bennyguo** [reply]
Thank you! Actually, our Fantasia3D implementation supports initializing from a custom mesh. We provide some results based on a human mesh [here](https://github.com/threestudio-project/threestudio#fantasia3d-) and you may try other custom meshes too. Looking forward to the article!

**Reply by jadevaibhav** [reply]
Along similar lines for NeRF-based models (DreamFusion etc), starting from some MLP checkpoint would accelerate convergence!

**Comment by yankeesong** [comment]
Implementation of [Magic123](https://arxiv.org/pdf/2306.17843.pdf). Most of the techniques are already implemented here so it might now be much more work.

**Comment by HL4214** [comment]
Implementation of DreamHuman. [DreamHuman: Animatable 3D Avatars from Text](https://dream-human.github.io/)

**Comment by ghost** [comment]
Implementation of [3DFuse : Let 2D Diffusion Model Know 3D-Consistency](https://arxiv.org/pdf/2303.07937.pdf) for Robust Text-to-3D Generation

project page: https://ku-cvlab.github.io/3DFuse/

![image](https://github.com/threestudio-project/threestudio/assets/97666314/7cd1a242-ce56-491f-b423-a4319958c82e)

**Reply by mskwak01** [reply]
Hello, I am one of the co-authors of 3DFuse, and we have an implementation of 3DFuse with ProlificDreamer baseline (other baselines can be easily adapted as well) with many promising results. Will this be incorporated if we push our code to Threestudio, or are there any further steps to take? Thank you!

**Comment by Njasa2k** [comment]
Requesting https://arxiv.org/pdf/2308.13223.pdf - EfficientDreamer: High-Fidelity and Robust 3D Creation
via Orthogonal-view Diffusion Prior
![this](https://github.com/threestudio-project/threestudio/assets/99513716/69823e8b-ac16-4afd-8a48-3a6eceae6217)

**Comment by TruthSearcher** [comment]
Implementation of
MVDream: Multi-view Diffusion for 3D Generation
https://arxiv.org/abs/2308.16512

**Comment by yuchenlichuck** [comment]
Implementation of 
Metrics to evaluate the qualitative of each model

**Comment by marinatenhave** [comment]
Huggingface or Tencent demo of ProlificDreamer, please!

**Comment by Njasa2k** [comment]
https://github.com/dreamgaussian/dreamgaussian implementation?

**Comment by TruthSearcher** [comment]
requesting Implementation for https://github.com/gsgen3d/gsgen

**Comment by Njasa2k** [comment]
can something like this be implemented 
![img](https://github.com/threestudio-project/threestudio/assets/99513716/76ecf19d-a63f-466d-bd70-c2297611476b)
like controlnet for 3d generations?
https://arxiv.org/abs/2310.05375 - IPDreamer: Appearance-Controllable 3D Object Generation with Image Prompts

**Comment by immaturehearts** [comment]
are u planning to implement _SyncDreamer_, please?

**Comment by zyms5244** [comment]
Implementation of RODIN Diffusion: https://3d-avatar-diffusion.microsoft.com/
![image](https://github.com/threestudio-project/threestudio/assets/10354474/76114802-9636-4567-830f-b320201528e9)
